src/api-functions/get-issues-by-project.ts

```diff
@@ -0,0 +1,101 @@
+import dotenv from "dotenv";
+
+export interface Issue {
+    project: string,
+    id: string,
+    summary: string,
+    description?: string,
+    assignee?: {
+      displayName: string,
+      accountId: string
+    },
+    reporter?: {
+      displayName: string,
+      accountId: string
+    },
+    priority: "Highest" | "High" | "Medium" | "Low" | "Lowest",
+    created: string,
+    duedate?: string,
+    timespent?: number
+};
+
+//Pulls Issue data of a given project and returns a promise with the data in a useable format
+export default async function getIssuesByProject(projectKey: string) : Promise<Issue[]> {
+    dotenv.config();
+    const username: string | undefined = process.env.ATLASSIAN_USERNAME;
+    const password: string | undefined  = process.env.ATLASSIAN_API_KEY;
+    const domain: string | undefined  = process.env.DOMAIN;
+    const userAuth = Buffer.from(`${username}:${password}`).toString('base64');
+
+    let isCompleted = false;
+    let result: Issue[] = [];
+    let startAt = 0;
+    
+    async function fetchIssues() {
+        const bodyData = `{
+            "fields": [
+                "summary",
+                "description",
+                "assignee",
+                "reporter",
+                "priority",
+                "duedate",
+                "created",
+                "timespent"
+            ],
+            "jql": "project = ${projectKey}",
+            "startAt": "${startAt}"
+        }`;
+        
+        const response = await fetch(`https://${domain}.atlassian.net/rest/api/3/search`, {
+            method: 'POST',
+            headers: {
+                'Authorization': `Basic ${userAuth}`,
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: bodyData
+        });
+
+        console.log(`Response: ${response.status} ${response.statusText}`);
+        const data = JSON.parse(await response.text());
+        
+        //Checks for full data pull
+        startAt = startAt+50;
```

Comment:
what is the use of 50 here.
Please dont use magic number.

Reply:
Please extract 50 into a variable and name it meaningfully.

Reply:
will your code pull all issues no matter how many issues are there or projects are there?

Reply:
Ok, made a resultsPerQuery variable that now equals 100, and added a "maxResults" property to the bodyData of the API call, so that 100 Issues are pulled per request. This number can be increased if needed, 50 is the default for an API call. You can see this in the new update.

Right now the code does pull all issues for a given project as it should. The implementation requires you to specify which project you want to pull from each time you run the program => `node app.js (databaseId) **(projectId)** `. This could be reworked to pulling from all available projects if you want.